### PR TITLE
Make tabulated shape families iterable

### PR DESCRIPTION
**Pull request by janbridley**
## Description

Tabulated shape families should be iterable to simplify code for users who wish to initialize every shape in a given family. I have also updated the documentation of the families to give a better tutorial.

Furthermore, I have deprecated the `PrismAntiprismFamily` and `PyramidDipyramidFamily` in favor of the simplified `Uniform*Family`, for new options [`Pyramid`, `Dipyramid`, `Prism`, `Antiprism`]. Aside from performance and precision improvements, the new implementations have a simplified interface for accessing the shapes. Previous version's are still implemented, but are no longer in the documentation.

~Readthedocs failing is not a result of this PR - doc builds work locally~ Readthedocs has been fixed

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

While the user-facing interface has remained the same, I changed a fair bit of code under the hood which could break for users with unusual scripts. For the most part, however, the API should be backwards compatible. I have removed the deprecated families from the documentation, but they are still functional as in previous versions.

### TODOS

- [x] Tests for new Prism and Antiprism families
    - [x] Based on previous code
    - [x] Based on shapes themselves
- [x] Translate Pyramid/Dipyramid families

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/coxeter/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/coxeter/blob/master/ContributorAgreement.md).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] I have updated the [changelog](https://github.com/glotzerlab/coxeter/blob/master/ChangeLog.txt) and the [credits](https://github.com/glotzerlab/coxeter/blob/master/doc/source/credits.rst).

### CODE REVIEW

**Comment by janbridley** [comment]
RTD builds are still failing - I will debug tomorrow.

File: coxeter/families/data/_previous_science1220869.json
Comment (janbridley) [inline coxeter/families/data/_previous_science1220869.json:1]: Previous data saved for posterity

File: coxeter/families/data/science1220869.json
Comment (janbridley) [inline coxeter/families/data/science1220869.json:1]: The new file is sorted, making the diff look very odd.

**Comment by DomFijan** [comment]
I can take a look at this in September.

**Comment by janbridley** [comment]
> I can take a look at this in September.

No rush - these are a nice to have more than anything

File: ChangeLog.rst [hunk 1]
```diff
@@ -4,6 +4,24 @@ This project adheres to `Semantic Versioning <http://semver.org/spec/v2.0.0.html
...
 -------------------
 
+Added
+~~~~~
+
+- ``TabulatedGSDShapeFamilies`` are now iterable, allowing easier access to shapes.
+- New ``UniformPrismFamily``, ``UniformAntiprismFamily``, ``UniformPyramidFamily``, and ``UniformDipyramidFamily``
+- New documentation to assist users initialize common geometries.
+
+Changed
```
Comment (DomFijan) [inline ChangeLog.rst:14]: Should there be a separate section with breaking changes only maybe?

**Review by DomFijan — APPROVED** [review]
Looks good! Docs also seem to render properly.

**Pull request merged**